feat(css/minifier): Handle nested `calc()`

ghostd opened the pull request:
<!-- Note: CI script will automatically rebase your PR so please do not rebase unless required -->

<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**Description:**

Handle nested `calc()`:
`calc(100% - calc(50% + 25px))` => `calc(50% + 25px)`

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->

**BREAKING CHANGE:**

<!--
If this PR introduces a breaking change, it must contain a notice for it to be included in the CHANGELOG. Add description or remove entirely if not breaking.

You may need to update `MIGRATION.md` for the breaking changes.
-->

**Related issue (if exists):**

alexander-akait commented on crates/swc_css_minifier/tests/fixture/compress-calc/discard-zero/output.min.css:
```diff
@@ -1 +1 @@
-.class1{width:calc(100vw/2 - 6px)}.class2{width:500px}.class3{width:calc(0 - 10px)}.class4{width:calc(0 - calc(1px + 1em))}.class5{width:calc(0 - (100vw - 10px)/2)}.class6{width:calc(10px - 100vw)}.class7{width:calc(0px - (100vw - 10px)*2)}.class8{width:calc(-10px - 100vw)}.class9{width:calc(0px - var(--foo,4px)/2)}
+.class1{width:calc(100vw/2 - 6px)}.class2{width:500px}.class3{width:calc(0 - 10px)}.class4{width:calc(0 - 1px - 1em)}.class5{width:calc(0 - (100vw - 10px)/2)}.class6{width:calc(10px - 100vw)}.class7{width:calc(0px - (100vw - 10px)*2)}.class8{width:calc(-10px - 100vw)}.class9{width:calc(0px - var(--foo,4px)/2)}
```
`width:calc(0 - 1px - 1em)` -> `width:calc(-1px - 1em)`

alexander-akait reviewed (changes requested):
nested `calc()` is like (` and `)` , so I think we should handle it not in `transform_calc_value_into_component_value`, we should handle them inside calculation.

ghostd commented:
@alexander-akait 'im not sure to understand your comment, i added the "nested calc" handling into the `resume` function. I rebased, i hope the diff will be more readable.

alexander-akait reviewed (approved):
Great, there are some impvomenets more, like `width:calc(2.25rem - 0px)`, but let's do it another PR, because it is not related to nested `calc`

alexander-akait commented:
@kdy1 Ready to review

kdy1 reviewed (approved):
Thanks!

---

swc-bump:
 - swc_css_minifier

ghostd commented:
@alexander-akait i'll work on removing zeroes this week.

alexander-akait commented:
Thanks
